srml/balances/src/lib.rs

```diff
@@ -399,11 +468,151 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
+/// there are more funds in the system than is accounted for in total issuance.
+#[must_use]
+pub struct NegativeImbalance<T: Subtrait<I>, I: Instance=DefaultInstance>(T::Balance);
+
+impl<T: Trait<I>, I: Instance> Imbalance<T::Balance> for PositiveImbalance<T, I> {
+	type Opposite = NegativeImbalance<T, I>;
+
+	fn zero() -> Self {
+		Self(Zero::zero())
+	}
+	fn drop_zero(self) -> result::Result<(), Self> {
+		if self.0.is_zero() {
+			Ok(())
+		} else {
+			Err(self)
+		}
+	}
+	fn split(self, amount: T::Balance) -> (Self, Self) {
+		let first = self.0.min(amount);
+		let second = self.0 - amount;
+		(Self(first), Self(second))
+	}
+	fn merge(self, other: Self) -> Self {
+		Self(self.0 + other.0)
+	}
+	fn subsume(&mut self, other: Self) {
+		self.0 += other.0
+	}
+	fn offset(self, other: Self::Opposite) -> result::Result<Self, Self::Opposite> {
+		if self.0 >= other.0 {
+			Ok(Self(self.0 - other.0))
+		} else {
+			Err(NegativeImbalance(other.0 - self.0))
+		}
+	}
+	fn peek(&self) -> T::Balance {
+		self.0.clone()
+	}
+}
+
+impl<T: Trait<I>, I: Instance> Imbalance<T::Balance> for NegativeImbalance<T, I> {
+	type Opposite = PositiveImbalance<T, I>;
+
+	fn zero() -> Self {
+		Self(Zero::zero())
+	}
+	fn drop_zero(self) -> result::Result<(), Self> {
+		if self.0.is_zero() {
+			Ok(())
+		} else {
+			Err(self)
+		}
+	}
+	fn split(self, amount: T::Balance) -> (Self, Self) {
+		let first = self.0.min(amount);
+		let second = self.0 - amount;
+		(Self(first), Self(second))
+	}
+	fn merge(self, other: Self) -> Self {
+		Self(self.0 + other.0)
+	}
+	fn subsume(&mut self, other: Self) {
+		self.0 += other.0
+	}
+	fn offset(self, other: Self::Opposite) -> result::Result<Self, Self::Opposite> {
+		if self.0 >= other.0 {
+			Ok(Self(self.0 - other.0))
+		} else {
+			Err(PositiveImbalance(other.0 - self.0))
+		}
+	}
+	fn peek(&self) -> T::Balance {
+		self.0.clone()
+	}
+}
+
+// Somewhat ugly hack in order to gain access to module's `increase_total_issuance_by`
+// using only the Subtrait (which defines only the types that are not dependent
+// on Positive/NegativeImbalance). Subtrait must be used otherwise we end up with a
+// circular dependency with Trait having some types be dependent on PositiveImbalance<Trait>
+// and PositiveImbalance itself depending back on Trait for its Drop impl (and thus
+// its type declaration).
+// This works as long as `increase_total_issuance_by` doesn't use the Imbalance
+// types (basically for charging feed).
+// This should eventually be refactored so that the three type items that do
+// depend on the Imbalance type (TransactionPayment, TransferPayment, DustRemoval)
+// are placed in their own SRML module.
+struct ElevatedTrait<T: Subtrait<I>, I: Instance>(T, I);
+impl<T: Subtrait<I>, I: Instance> Clone for ElevatedTrait<T, I> {
+	fn clone(&self) -> Self { unimplemented!() }
+}
+impl<T: Subtrait<I>, I: Instance> PartialEq for ElevatedTrait<T, I> {
+	fn eq(&self, _: &Self) -> bool { unimplemented!() }
```

Comment:
These config trait types are never instantiated and indeed not designed for that; i'd prefer for them to actually panic to denote that rather than attempting a meaningless impl.